packages/core/src/layout/GridItem/GridItem.tsx

```diff
@@ -40,9 +45,14 @@ const colEnd = "auto";
 const rowStart = "auto";
 const rowEnd = "auto";
 
-export const GridItem = forwardRef<HTMLDivElement, GridItemProps>(
-  function GridItem(
+type GridItemComponent = <T extends ElementType = "div">(
+  props: GridItemProps<T>
+) => ReactElement | null;
+
+export const GridItem: GridItemComponent = forwardRef(
+  <T extends ElementType = "div">(
     {
+      as,
```

Comment:
do we limit the possible tags that can be used as `Component`? should we?

Reply:
At the moment we accept any valid `ElementType` but it's open for discussion although I think if we come up with a list it would be quite long